manual/node_pattern.md

```diff
@@ -0,0 +1,270 @@
+# Node Pattern
```

Comment:
I think this should tutorial should begin with a description of the problem node patterns were created to solve - a few examples of "legacy code" that had to do multiple node destructurings to access some node(s) and the same code expressed using node patterns. You can find plenty of examples of this in RuboCop's own codebase - after the node pattern was introduced by @alexdowad we've started gradually porting cops to it. Just look at the git history.

Reply:
Good point. I'll get a few examples and use it!
I directed to the details of the language, but it's an excellent addition to understanding why it was built.

I'll certainly take a look at the codebase. Do you remember any extraordinary example we can show the "before" and "after" using the Node Pattern?